release-20.1: builtins: fix bug in extract epoch from timestamptz in non-UTC timezone

otan opened the pull request:
Backport 1/1 commits from #50071.

/cc @cockroachdb/release

---

Release note (bug fix): `extract(epoch from timestamptz)` from a session
time zone which is not in UTC would previously return a value
which has been incorrect offset by the session time zone. This has now
been rectified.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/50075)
<!-- Reviewable:end -->

RaduBerinde reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/50075#-:-M9WWSIL4WZk2qogML3a:bnfp4nl)* status: :shipit: complete! 1 of 0 LGTMs obtained (waiting on @RaduBerinde)



<!-- Sent from Reviewable.io -->

otan committed:
builtins: fix bug in extract epoch from timestamptz in non-UTC timezone

Release note (bug fix): `extract(epoch from timestamptz)` from a session
time zone which is not in UTC would previously return a value
which has been incorrect offset by the session time zone. This has now
been rectified.
